### DISCUSSION TITLE
Is it possible to pipe the formatted content of bat to a file?

### DISCUSSION THREAD

**Discussion by jmcharter** [sharkdp/bat#2746] [Q&A] [answered]
I'm trying to pipe the nicely formatted content (including line numbers and file names) to a file

`bat path/to/dir/* | file.txt`

But this only pipes the plain text without file names. I can see why that would typically be desirable, but is there a way to include everything?

**Comment by keith-hall** [comment]
You could try explicitly setting the style so it doesn't resort to plain while piping https://github.com/sharkdp/bat#output-style

**Reply by jmcharter** [reply]
Hi Keith

Thanks for the suggestion. I did already try that, and unfortunately the styling gets stripped.

**Comment by einfachIrgendwer0815** [comment, marked as answer]
`--decorations=always` should include it, I guess.

**Reply by jmcharter** [reply]
Yes, this worked. I missed this option when I was looking, but the description is perfectly adequate. Thank you.
